packages/edit-site/src/components/global-styles/utils.js

```diff
@@ -255,3 +255,34 @@ export function getValueFromVariable( features, blockName, variable ) {
 	}
 	return variable;
 }
+
+/**
+ * Function that scopes a selector with another one. This works a bit like
+ * SCSS nesting except the `&` operator isn't supported.
+ *
+ * @example
+ * ```js
+ * const scope = '.a, .b .c';
+ * const selector = '> .x, .y';
+ * const merged = scopeSelector( scope, selector );
+ * // merged is '.a > .x, .a .y, .b .c > .x, .b .c .y'
+ * ```
+ *
+ * @param {string} scope    Selector to scope to.
+ * @param {string} selector Original selector.
+ *
+ * @return {string} Scoped selector.
+ */
+export function scopeSelector( scope, selector ) {
```

Comment:
Not a biggie for this PR but someday it'd be nice to put this one somewhere where duotone and global styles could access. 

I say "nice" but it's hard one to categorize 🤔 

I see we export `useCustomSides` from '@wordpress/block-editor' and use it in the [edit-side package.](https://github.com/WordPress/gutenberg/blob/3f3c8df23c70a37b7ac4dddebc82030362133593/packages/edit-site/src/components/global-styles/dimensions-panel.js#L12), but do you think we could/should do something similar for `scopeSelector()`and add it to the block-editor package API?

On the surface it looks like a general styles utility that might one day live in the style engine.

Reply:
This was one I meant to come back to and find a proper home for. 

> On the surface it looks like a general styles utility that might one day live in the style engine.

It is most relevant to the generation of styles so the style engine is probably the best long-term candidate. I wasn't sure whether it would be ok to add it to the block-editor package API or if that would be committing us to a deprecation cycle later on if it gets rolled into the style engine.

I don't feel strongly about it so happy to proceed if you'd recommend that approach.

Reply:
> I don't feel strongly about it so happy to proceed if you'd recommend that approach.

Me either 😄 

Maybe one to defer since it's doesn't really deal with the main thrust of the PR 👍